src/Widgets/AgendaEventRow.vala

```diff
@@ -315,15 +317,24 @@ public class Maya.View.AgendaEventRow : Gtk.ListBoxRow {
                     datatime_label.hide ();
                     datatime_label.no_show_all = true;
                 } else {
-                    // TRANSLATORS: A range from start time to end time i.e. "7:00 PM – 9:00 PM"
-                    datetime_string = C_("time-range", "%s – %s").printf (start_time_string, end_time_string);
+                    if (is_same_time) {
```

Comment:
So I actually have a webcal that has an event which is a deadline and for some reason even with this branch it's reading as `4:00 pm – 4:00pm`. I wonder if we should compare the start and end time strings and not the actual datetimes. I wonder if the end datetime has like a null date or something weird

Reply:
I changed it to compare strings and it works on my previous events. :crossed_fingers: